modules/platform/nuxeo-drive-server/nuxeo-drive-mongodb/src/main/java/org/nuxeo/drive/mongodb/MongoDBAuditChangeFinder.java

```diff
@@ -0,0 +1,156 @@
+import static com.mongodb.client.model.Filters.exists;
+import static com.mongodb.client.model.Filters.gt;
+import static com.mongodb.client.model.Filters.in;
+import static com.mongodb.client.model.Filters.lte;
+import static com.mongodb.client.model.Filters.ne;
+import static com.mongodb.client.model.Filters.or;
+import static com.mongodb.client.model.Filters.regex;
+import static com.mongodb.client.model.Sorts.ascending;
+import static com.mongodb.client.model.Sorts.descending;
+import static com.mongodb.client.model.Sorts.orderBy;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_ID;
+import static org.nuxeo.ecm.platform.audit.api.BuiltinLogEntryData.LOG_LOG_DATE;
+
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.nuxeo.drive.service.SynchronizationRoots;
+import org.nuxeo.drive.service.impl.AuditChangeFinder;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.ecm.core.query.sql.model.OrderByExprs;
+import org.nuxeo.ecm.core.query.sql.model.Predicates;
+import org.nuxeo.ecm.core.query.sql.model.QueryBuilder;
+import org.nuxeo.ecm.platform.audit.api.AuditQueryBuilder;
+import org.nuxeo.ecm.platform.audit.api.AuditReader;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.mongodb.audit.MongoDBAuditBackend;
+import org.nuxeo.mongodb.audit.MongoDBAuditEntryReader;
+import org.nuxeo.runtime.api.Framework;
+
+import com.mongodb.client.MongoCollection;
+
+/** @since 11.3 */
+public class MongoDBAuditChangeFinder extends AuditChangeFinder {
+
+    private static final Logger log = LogManager.getLogger(MongoDBAuditChangeFinder.class);
+
+    @Override
+    public long getUpperBound() {
+        RepositoryManager repositoryManager = Framework.getService(RepositoryManager.class);
+        return getUpperBound(new HashSet<>(repositoryManager.getRepositoryNames()));
+    }
+
+    /**
+     * Returns the last available log id in the audit collection considering events older than the last clustering
+     * invalidation date if clustering is enabled for at least one of the given repositories. This is to make sure the
+     * {@code DocumentModel} further fetched from the session using the audit entry doc id is fresh.
+     */
+    @Override
+    public long getUpperBound(Set<String> repositoryNames) {
+        long clusteringDelay = getClusteringDelay(repositoryNames);
+        MongoDBAuditBackend auditService = (MongoDBAuditBackend) Framework.getService(AuditReader.class);
+        QueryBuilder queryBuilder = new AuditQueryBuilder().predicate(
+                Predicates.gt("eventDate", new Date(0)));
```

Comment:
Could you please add a comment explaining why we do this, because we don't have APIs to do a query with no predicate at all? Like `// TODO remove this dummy predicate once we can query with no predicate at all`